devito/archinfo.py

```diff
@@ -107,15 +113,31 @@ def get_cpu_brand():
         # Fallback 1: it should now be fine to use psutil
         physical = psutil.cpu_count(logical=False)
         if not physical:
-            # Fallback 2: we might end up here on more exotic platforms such a Power8
-            # Hopefully we can rely on `lscpu`
             try:
-                physical = lscpu()['Core(s) per socket'] * lscpu()['Socket(s)']
+                lscpu_mapper = lscpu()
+                # Fallback 2: we might end up here
+                # on more exotic platforms such a Power8 or due to
+                # erroneous autodetection
+                physical = (lscpu_mapper['Core(s) per socket']
+                            * lscpu_mapper['Socket(s)'])
             except KeyError:
                 warning("Physical core count autodetection failed")
                 physical = 1
-    cpu_info['physical'] = physical
 
+    try:
```

Comment:
this entire try-except should be moved up... 

probably around [here](https://github.com/devitocodes/devito/blob/master/devito/archinfo.py#L85)

then we update `cpu_info` and return without going through the generic detection code. After all, this is a special case, and the special cases should be dealt with **before** the general case, almost always

Reply:
Moved up, hope its fine.